Rakefile

```diff
@@ -29,35 +27,57 @@ task :check_for_node do
     return
   end
 
-  unless system(which, 'node', opts)
+  unless program_exists? 'node'
     abort 'Please install Node.js: https://nodejs.org/'
   end
-
-  programs = [
-    'bower',
-    'bower-installer',
-    ['uglify-js', 'uglifyjs'],
-  ]
-  programs.each do |npm|
-    pkg, prog = npm
-    prog = npm if prog == nil
-    unless program_exists? prog
-      abort "Please install #{pkg}: npm install -g #{pkg}"
-    end
-  end
 end
 
 desc "Update JavaScript components"
-task :update_js => :check_for_node do
-  abort unless system('bower', 'update')
-  abort unless system('bower-installer')
+task :update_js_components => :check_for_node do
+  abort unless system 'npm', 'update'
+  abort unless system 'bower', 'update'
 end
 
-desc "Compress JavaScript components"
-task :compress_js => :check_for_node do
-  Dir[File.join(File.dirname(__FILE__), 'assets', 'js', '**', '*')].each do |f|
-    next unless File.file? f and f.end_with? '.js' and not f.end_with? '.min.js'
-    outfile = "#{f[0..-('.js'.size + 1)]}.min.js"
-    abort unless system('uglifyjs', '-c', '-m', '--', f, :out=>outfile)
+LIB_LUNR_TARGET = File.join %w(lib jekyll_pages_api_search lunr.min.js)
+LIB_LUNR_SOURCE = File.join %w(bower_components lunr.js lunr.min.js)
+
+file LIB_LUNR_TARGET => LIB_LUNR_SOURCE do
+  FileUtils.cp LIB_LUNR_SOURCE, LIB_LUNR_TARGET
+end
+
+# TODO(mbland): Extract this and other bits from this Rakefile into a gem.
+cxt = V8::Context.new
+basedir = File.dirname(__FILE__)
+package_json = File.read File.join(basedir, 'package.json')
+cxt.eval "var package = #{package_json};"
+main_js = cxt[:package].main
+search_bundle = File.join 'assets', 'js', 'search-bundle.js'
+
+file search_bundle => main_js do
+  unless system 'browserify', '-g', 'uglifyify', main_js, :out=>search_bundle
+    abort "browserify failed"
   end
 end
+
+search_bundle_gz = "#{search_bundle}.gz"
+
+file search_bundle_gz => search_bundle do
+  unless program_exists? 'gzip'
+    puts "Cannot determine if the gzip program exists on the system; " +
+      "skipping compression."
+    return
+  end
+  unless system 'gzip', '--best', '-c', search_bundle, :out=>search_bundle_gz
+    abort "compression failed for: #{search_bundle}"
+  end
+end
+
+task :build_js => [:check_for_node, LIB_LUNR_TARGET, search_bundle] do
+end
+
+task :test => :build_js
+
+task :build => [
+  :test,
+  search_bundle_gz,
+]
```

Comment:
Not a blocker, but I wonder if we might want to look into moving this stuff to Grunt/Gulp/etc. since all the operations are in Node-land anyway.